### ISSUE TITLE
First periodic box vector must be parallel to X - error

### ISSUE THREAD

**Issue by praban07** [openmm/openmm#4454]
I am running a PMF calculation. For the first point, it runs fine. However, for the second point (which has the same starting point as the first point) I am getting the error
"First periodic box vector must be parallel to X"

Although I see that this error has been encountered by other people (and solved it), I am not sure why is it coming here. I have attached the code and all the relevant files. Can anyone suggest what is happening?

[err.tar.gz](https://github.com/openmm/openmm/files/14418972/err.tar.gz)

**Comment by peastman** [comment]
I think there's something wrong with your pressure regulation.  The log shows that the box volume starts out at 94, then grows to over 80,000!  That suggests something in your forces is applying a huge outward pressure, forcing the box to expand.

**Comment by praban07** [comment]
You are right. It is coming from the CV force. However, the intriguing part is that the same code runs in the other two machines of my lab, one of which has the same GPU card (RX3090) as in the machine (say, computer A) where it is not working. Any idea what could be the reason?

**Comment by peastman** [comment]
The GPU shouldn't make any difference.  Is the box size also growing on the other machines?

**Comment by praban07** [comment]
I could pinpoint the issue. It starts with the energy calculation. 
It runs FINE in other machines but has problem in this machine. Here is the relevant part of the code (this was done with your suggestions and works fine in all other machines where I have checked). Just to summarize, we copy the original nonbonded to el_term_1 (also to el_term_2, which I am not showing below). Then original nonbonded calculates only LJ and el_term_1 and el_term_2 are used to define the CV force (not showing the exceptions used for el_term_1 in the code below). However, el_term_1 gives incorrect result in this particular machine. The whole code can be seen from the zip file I attached previously. Any suggestions?
````````````````````````
el_term_1=copy.copy(nonbonded)
# the following block ensures that the original nonbonded force calculates only
# LJ (after scaling), not the electrostatics for all particles. All charges are set to zero.


for index in range(nonbonded.getNumParticles()):
    charge, sigma, epsilon = nonbonded.getParticleParameters(index)
#    print('charge',charge,sigma,epsilon,index)
    nonbonded.setParticleParameters(index, charge*zero, sigma, epsilon)
#    nonbonded.setParticleParameters(index, charge, sigma, epsilon)

#nonbonded.setForceGroup(1)
scale_map =  dict(zip(scale_indices, scale_factors))

# putting exceptions in the nonbonded force with scaled chargeProd

for i in range(nonbonded.getNumExceptions()):
    p1, p2, chargeProd, sigma, epsilon = nonbonded.getExceptionParameters(i)
#    print('p1p2-all',p1,p2)
# is either p1 or p2 or both are present, we scale the chargeprod
    if p1 in scale_indices or p2 in scale_indices:
#        print('p1p2',p1,p2)
        chargeProd *= scale_factors[0]
    nonbonded.setExceptionParameters(i, p1, p2, chargeProd, sigma, epsilon)


# el_term_1 calculates only electrostatics for all particles
for index in range(el_term_1.getNumParticles()):
    charge, sigma, epsilon = el_term_2.getParticleParameters(index)
    epsilon=0

    el_term_1.setParticleParameters(index, charge, sigma, epsilon)
``````````````````````````````````````````````````````

**Comment by peastman** [comment]
When you say the `el_term_1` term is incorrect, what exactly does that mean?  What energy does it compute?  What should it be?

Which platform are you using?  Does the result change if you use a different platform?
